api/envoy/config/bootstrap/v3/bootstrap.proto

```diff
@@ -250,7 +250,20 @@ message Bootstrap {
   // Setting this value causes failure if the
   // ``envoy.restart_features.use_apple_api_for_dns_lookups`` runtime value is true during
   // server startup. Apple' API only uses UDP for DNS resolution.
-  bool use_tcp_for_dns_lookups = 20;
+  //
+  // .. attention::
+  //
+  //   This field is deprecated in favor of *dns_resolver_options*
+  //   which aggregates all of the DNS resolver options in a single message.
+  bool use_tcp_for_dns_lookups = 20
+      [deprecated = true, (envoy.annotations.deprecated_at_minor_version) = "3.0"];
+
+  // Configuration of DNS resolver option flags which control the behavior of the DNS resolver.
+  // This may be overridden on a per-cluster basis in cds_config,
+  // when :ref:`dns_resolvers <envoy_api_field_config.cluster.v3.Cluster.dns_resolvers>` and
+  // :ref:`dns_resolver_options <envoy_api_field_config.cluster.v3.Cluster.dns_resolver_options>`
+  // are specified.
```

Comment:
I'm not sure I understand this comment.  Can you expand on this a bit more?

Reply:
I copied this from [envoy docs](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/bootstrap/v3/bootstrap.proto.html) "cmd+f" **This may be overridden on a per-cluster basis in ...** description of _use_tcp_for_dns_lookups_.